### PR TITLE
Update multihash dependency and make Cid impl Copy

### LINKED ISSUE #837: Update multihash dependency
Multihash dependency need to be updated apparently.

Here the report from crate audit
```
error: Vulnerable crates found!

ID:       RUSTSEC-2020-0068
Crate:    multihash
Version:  0.10.1
Date:     2020-11-08
URL:      https://rustsec.org/advisories/RUSTSEC-2020-0068
Title:    Unexpected panic in multihash `from_slice` parsing code
Solution:  upgrade to >= 0.11.3
Dependency tree: 
multihash 0.10.1
└── forest_cid 0.1.1
    ├── forest_vm 0.2.1
    │   ├── forest_message 0.5.0
    │   │   ├── hfuzz-signer 0.0.0
    │   │   ├── filecoin-signer 0.1.0
    │   │   │   ├── hfuzz-signer 0.0.0
    │   │   │   ├── filecoin-signer-wasm 0.1.0
    │   │   │   ├── filecoin-signer-ffi 0.1.0
    │   │   │   └── filecoin-service 0.1.0
    │   │   ├── filecoin-service 0.1.0
    │   │   └── extras 0.1.0
    │   │       └── filecoin-signer 0.1.0
    │   ├── filecoin-signer 0.1.0
    │   └── extras 0.1.0
    ├── forest_message 0.5.0
    ├── forest_encoding 0.1.2
    │   ├── forest_vm 0.2.1
    │   ├── forest_message 0.5.0
    │   ├── forest_crypto 0.3.1
    │   │   ├── hfuzz-signer 0.0.0
    │   │   ├── forest_message 0.5.0
    │   │   ├── filecoin-signer 0.1.0
    │   │   └── extras 0.1.0
    │   ├── forest_address 0.2.3
    │   │   ├── hfuzz-signer 0.0.0
    │   │   ├── forest_vm 0.2.1
    │   │   ├── forest_message 0.5.0
    │   │   ├── forest_crypto 0.3.1
    │   │   ├── filecoin-signer 0.1.0
    │   │   └── extras 0.1.0
    │   ├── filecoin-signer 0.1.0
    │   └── extras 0.1.0
    ├── filecoin-signer 0.1.0
    └── extras 0.1.0
```

### PR DESCRIPTION
**Pull request by austinabell**
**Summary of changes**
Changes introduced in this pull request:
- Updating due to security vulnerability. Was a decent amount of breaking changes but now that the alloc size is limited (and our max is 32 bytes afaik unless there is an identity hash that rust impl allows to be greater), `Cid` can now be clone

cc @vmx let me know if you think anything should be changed from this :)

**Reference issue to close (if applicable)**

Closes #837 

**Other information and links**

### CODE REVIEW

**Comment by ec2** [comment]
You mean Cid can now be copy?

File: blockchain/chain/Cargo.toml [hunk 1]
```diff
@@ -36,7 +36,7 @@ interpreter = { path = "../../vm/interpreter/" }
 json = []
 
 [dev-dependencies]
-multihash = "0.10.0"
+multihash = { version = "0.13", default-features = false, features = ["std", "multihash-impl", "blake2b", "sha2", "multihash-impl", "derive"] }
```
Comment (vmx) [inline blockchain/chain/Cargo.toml:39]: I don't think you need `multihash-impl` here as you derive your own table (same for the `Cargo.toml` further down.

**Review by vmx — COMMENTED** [review]
Looks good, it matches how I hoped this version would be used :)

File: blockchain/chain/Cargo.toml (hunk 1 above)
Comment (austinabell) [reply blockchain/chain/Cargo.toml]: ah ya I forgot to clean up the features, also don't need sha2, thanks for pinging

**Review by ec2 — APPROVED** [review]

**Review by dutterbutter — APPROVED** [review]

**Pull request merged**